### DISCUSSION TITLE
Branch Owners instead of Code Owners

### DISCUSSION THREAD

**Discussion by Danielku15** [community/community#30644] [Pull Requests]
Hi. 

We are aiming for a setup where we have mandatory reviewers for specific branches as part of our repository branch protections. Unfortunately codeowners definition currently lives in the code making it hard to define proper restrictions. There are quite some problems with managing this required reviewers through this file in code: 

1. You cannot properly manage those settings centrally easily. We use the GitHub APIs to roll-out unified settings to our repos. But maintaining codeowners files just doesn't seem feasable to do through APIs or Actions. 
2. When starting branches you automatically also inherit the codeowners which is not what we want. e.g. you start a feature branch from main. You inherit the codeowners which is not true for this new branch. But also when merging between branches it can easily happen that codeowners are overwritten unless you work with cherrypicking or again special merge branches where you undo the codeowners change. In our case we have a different ownership on `develop` (multiple dev teams) than on `main` (one team responsible for the releases).

In mid term I would hope that we can define as part of branch protections the mandatory reviewers as explicit list (like the other exceptions/exclusions). Hence this would be a feature request but I could not find a discussion (yet) for this. Has anyone heard of such a feature being requested and maybe even developed in GitHub?

We also somehow need a temporary custom solution for that, hence I am wondering if anyone has maybe build such a restriction using GitHub Actions. I could imagine a custom action could also block the merging until a 'branch' owner of the PR target branch approved the PR. Has anyone experience there or even a working solution? 

Greetings
Daniel

**Comment by wavded** [comment]
Find any solutions/workarounds for this? Would be great to have `codeowners` file be able to specify the branch it should apply to.
